easybuild/easyblocks/generic/tarball.py

```diff
@@ -57,8 +76,29 @@ def build_step(self):
 
     def install_step(self, src=None):
         """Install by copying from specified source directory (or 'start_dir' if not specified)."""
-        remove_dir(self.installdir)
-        copy_dir(src or self.cfg['start_dir'], self.installdir, symlinks=self.cfg['keepsymlinks'])
+
+        # Run preinstallopts and/or preinstall_cmd before copy of source directory
+        preinstall_cmd = None
+        if self.cfg['preinstallopts']:
+            preinstall_opts = self.cfg['preinstallopts'].split('&&')
+            preinstall_cmd = '&&'.join([opt for opt in preinstall_opts if opt and not opt.isspace()])
+        if self.cfg['preinstall_cmd']:
+            preinstall_cmd = '&& '.join([cmd for cmd in [preinstall_cmd, self.cfg['preinstall_cmd']] if cmd])
+        if preinstall_cmd:
+            self.log.info("Preparing installation of %s using command '%s'..." % (self.name, preinstall_cmd))
+            run_cmd(preinstall_cmd, log_all=True, simple=True)
+
+        # Copy source directory
+        source_path = src or self.cfg['start_dir']
+
+        if self.cfg['install_in_subdir']:
+            install_path = os.path.join(self.installdir, self.name.lower())
+        else:
+            install_path = self.installdir
+
+        self.log.info("Copying tarball contents of %s to %s..." % (self.name, install_path))
+        remove_dir(install_path)
+        copy_dir(source_path, install_path, symlinks=self.cfg['keepsymlinks'])
 
```

Comment:
Should we remove dir? shouldn't easyblock wipe whole installdir in the begining?

copy_dir(source_path, install_path, symlinks=self.cfg['keepsymlinks'], dirs_exist_ok=self.cfg['dirs_exist_ok'])

Reply:
Removing the directory is still required I think if this easyblock is used outside a bundle, but it would now need to be done conditionally, something like 
```
dirs_exist_ok = self.cfg['dirs_exist_ok']
if not dirs_exist_ok:
    remove_dir(install_path)
copy_dir(source_path, install_path, symlinks=self.cfg['keepsymlinks'], dirs_exist_ok=dirs_exist_ok)
```

Reply:
I removed the `remove_dir()` call if `tarball` is used in merge mode.